### PR TITLE
Make nox pass again

### LINKED ISSUE #74: nox is failing
**Describe the bug**
Running `nox` is failing.

**To Reproduce**
Run `nox`.

**Expected behavior**
All tests pass.

Fix this.

### PR DESCRIPTION
**Pull request by ocelotl**
Fixes #74

This makes `nox` pass again, which is a basic need for further development. I have also fixed the test requirements to shield us against sudden failures that may happen because a dependency is upgraded.

### CODE REVIEW

**Review by codeboten — COMMENTED** [review]
Is this all required to get nox to pass? The change looks much bigger than this.

**Comment by ocelotl** [comment]
> Is this all required to get nox to pass? The change looks much bigger than this.

Iep, for `nox` to pass we had to get rid of any reference to metrics because having them will cause `ImportError`s and fail the whole runs. Also, we have dependencies that no longer support 3.5 so this version had to be removed.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 ## Unreleased
 
+- Removing metrics configuration
```
Comment (codeboten) [inline CHANGELOG.md:3]: please add a note regarding python 3.5 deprecation

Comment (codeboten) [inline CHANGELOG.md:3]: Can this be updated to make mention of this being removed as it is not part of trace GA

File: setup.cfg [hunk 2]
```diff
@@ -40,12 +38,11 @@ package_dir=
 packages=find_namespace:
 install_requires =
     environs >= 8.0.0
-    opentelemetry-api == 0.17b0
-    opentelemetry-sdk == 0.17b0
-    opentelemetry-exporter-otlp == 0.17b0
-    opentelemetry-propagator-b3 == 0.17b0
-    opentelemetry-instrumentation == 0.17b0
-    opentelemetry-instrumentation-system-metrics == 0.17b0
+    opentelemetry-api
```
Comment (codeboten) [inline setup.cfg:41]: please set the version number here

File: requirements-test.txt [hunk 3]
```diff
@@ -1,5 +1,45 @@
...
-pytest-cov==2.9.0
-ipdb
-opentelemetry-instrumentation-requests
-opentelemetry-instrumentation-flask
+aiocontextvars==0.2.2
+asgiref==3.3.1
+attrs==20.3.0
+backoff==1.10.0
+certifi==2020.12.5
+chardet==4.0.0
```
Comment (codeboten) [inline requirements-test.txt:6]: are all these required?

**Review by codeboten — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (ocelotl) [reply CHANGELOG.md:3]: Fixed

File: setup.cfg (hunk 2 above)
Comment (ocelotl) [reply setup.cfg:41]: Fixed

File: CHANGELOG.md (hunk 1 above)
Comment (ocelotl) [reply CHANGELOG.md]: Fixed

**Review by codeboten — APPROVED** [review]

**Pull request merged**